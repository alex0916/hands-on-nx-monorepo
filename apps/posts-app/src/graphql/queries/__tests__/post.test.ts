import { getTestServer } from '@gql/test-server';
import { POSTS } from '@gql/test-operations';
import { postsFixture, commentsFixture, usersFixture } from '@gql/test-fixtures';

const { mockContext, executeOperation } = getTestServer();

const subject = async (variables: Record<string, unknown>) =>
	executeOperation(
		{
			query: POSTS,
			variables,
		},
		{ contextValue: mockContext }
	);

afterEach(() => {
	jest.resetAllMocks();
});

describe('Posts queries', () => {
	it('should resolve the posts', async () => {
		const { postService, commentService, userService } = mockContext.dataSources;
		postService.getPosts = jest.fn().mockResolvedValue(postsFixture);

		commentService.getCommentsByPostId = jest.fn((id: string) =>
			Promise.resolve(commentsFixture.filter(({ postId }) => postId === id))
		);

		userService.getUserById = jest.fn((userId: string) =>
			Promise.resolve(usersFixture.find(({ id }) => id === userId))
		);

		const { data, errors } = await subject({ first: 2 });
		expect(errors).toBeUndefined();
		expect(data).toMatchSnapshot();
	});
});
