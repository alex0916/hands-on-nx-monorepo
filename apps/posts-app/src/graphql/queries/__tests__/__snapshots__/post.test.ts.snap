// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts queries should resolve the posts 1`] = `
Object {
  "posts": Object {
    "edges": Array [
      Object {
        "cursor": "post1",
        "node": Object {
          "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
          "id": "post1",
          "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
          "totalComments": 1,
          "user": Object {
            "avatar": "https://api.dicebear.com/5.x/avataaars/svg?seed=test1",
            "email": "Sincere@april.biz",
            "id": "test1",
            "name": "Leanne Test",
          },
        },
      },
      Object {
        "cursor": "post2",
        "node": Object {
          "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
          "id": "post2",
          "title": "qui est esse",
          "totalComments": 1,
          "user": Object {
            "avatar": "https://api.dicebear.com/5.x/avataaars/svg?seed=test1",
            "email": "Sincere@april.biz",
            "id": "test1",
            "name": "Leanne Test",
          },
        },
      },
    ],
    "pageInfo": Object {
      "endCursor": "post2",
      "hasNextPage": true,
      "hasPreviousPage": false,
      "startCursor": "post1",
    },
  },
}
`;
